### PR TITLE
Ford CAN FD safety

### PR DESCRIPTION
**Pull request by incognitojam**

### CODE REVIEW

File: board/safety/safety_ford.h [hunk 1]
```diff
@@ -364,10 +420,10 @@ static int ford_fwd_hook(int bus_num, int addr) {
 }
 
 static const addr_checks* ford_init(uint16_t param) {
-  UNUSED(param);
 #ifdef ALLOW_DEBUG
   ford_longitudinal = GET_FLAG(param, FORD_PARAM_LONGITUDINAL);
 #endif
+  ford_canfd = GET_FLAG(param, FORD_PARAM_CANFD);
```
Comment (adeebshihadeh) [inline board/safety/safety_ford.h:426]: move this behind the debug flag

File: board/safety/safety_ford.h [hunk 2]
```diff
@@ -154,12 +178,16 @@ const LongitudinalLimits FORD_LONG_LIMITS = {
 #define INACTIVE_CURVATURE_RATE 4096U
 #define INACTIVE_PATH_OFFSET 512U
 #define INACTIVE_PATH_ANGLE 1000U
+
+#define CANFD_INACTIVE_CURVATURE_RATE 1024U
```
Comment (adeebshihadeh) [inline board/safety/safety_ford.h:182]: should be namespaced with FORD_

**Review by adeebshihadeh — APPROVED** [review]
Good to merge once comments are addressed and tests are passing

File: board/safety/safety_ford.h [hunk 3]
```diff
@@ -327,6 +362,27 @@ static int ford_tx_hook(CANPacket_t *to_send) {
...
+
+    // Check angle error and steer_control_enabled
+    int desired_curvature = raw_curvature - FORD_INACTIVE_CURVATURE;  // /FORD_STEERING_LIMITS.angle_deg_to_can to get real curvature
+    violation |= steer_angle_cmd_checks(desired_curvature, steer_control_enabled, FORD_STEERING_LIMITS);
+
+    if (violation) {
+      tx = 0;
+    }
+  }
+
```
Comment (sshane) [inline board/safety/safety_ford.h:385]: Can we not combine this with the existing check?

File: tests/safety/test_ford.py [hunk 4]
```diff
@@ -158,16 +167,26 @@ def _lkas_command_msg(self, action: int):
...
+      return self.packer.make_can_msg_panda("LateralMotionControl", 0, values)
+    elif self.STEER_MESSAGE == MSG_LateralMotionControl2:
+      values = {
+        "LatCtl_D2_Rq": 1 if enabled else 0,
+        "LatCtlPathOffst_L_Actl": path_offset,     # Path offset [-5.12|5.11] meter
+        "LatCtlPath_An_Actl": path_angle,          # Path angle [-0.5|0.5235] radians
+        "LatCtlCrv_NoRate2_Actl": curvature_rate,  # Curvature rate [-0.001024|0.001023] 1/meter^2
+        "LatCtlCurv_No_Actl": curvature,           # Curvature [-0.02|0.02094] 1/meter
+      }
+      return self.packer.make_can_msg_panda("LateralMotionControl2", 0, values)
```
Comment (sshane) [inline tests/safety/test_ford.py:189]: Think it would be more in-convention if you just override this function in the CAN FD test

**Pull request merged**